src/h5web/visualizations/heatmap/HeatmapVis.tsx

```diff
@@ -30,24 +36,66 @@ function HeatmapVis(props: Props): JSX.Element {
     keepAspectRatio = true,
     showGrid = false,
     showLoader = true,
+    abscissas,
+    ordinates,
   } = props;
 
   const { rows, cols } = getDims(dataArray);
   const aspectRatio = keepAspectRatio ? cols / rows : undefined; // width / height <=> cols / rows
 
+  const abscissaIndices = range(cols + 1);
+  const abscissaValues = abscissas || abscissaIndices;
+  const abscissaToIndex = getValueToIndexScale(abscissaValues);
+  const abscissaDomain = useMemo(() => {
+    return getDataDomain(abscissaValues) ?? DEFAULT_DOMAIN;
+  }, [abscissaValues]);
+
+  const ordinateIndices = range(rows + 1);
+  const ordinateValues = ordinates || ordinateIndices;
+  const ordinateToIndex = getValueToIndexScale(ordinateValues);
+  const ordinateDomain = useMemo(() => {
+    return getDataDomain(ordinateValues) ?? DEFAULT_DOMAIN;
+  }, [ordinateValues]);
+
+  if (abscissaValues.length !== cols + 1) {
```

Comment:
`abscissaValues` and `ordinateValues` give the positions of the pixels' edges. For `N` pixels, we have `N+1` edges.